src/tools/tidy/src/cargo_lock.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::path::Path;
+
+fn filter_lock_files(path: &Path) -> bool {
+    use std::fs;
+
+    match fs::metadata(path) {
+        Ok(m) => {
+            if !m.is_dir() {
+                return true;
+            }
+            let dir = fs::read_dir(path);
+            for entry in t!(dir).map(|e| t!(e)) {
+                if entry.file_name().to_str() == Some("Cargo.lock") {
+                    return false;
+                }
+            }
+            return true;
+        },
+        Err(_) => { true }
+    }
+}
+
+pub fn check(path: &Path, bad: &mut bool) {
+    use std::process::Command;
+
+    super::walk(path,
+                &mut |path| filter_lock_files(path),
+                &mut |_| {
+                    let ret_code = Command::new("git")
+                                                .arg("diff-index")
+                                                .arg("--quiet")
+                                                .arg("HEAD")
+                                                .arg("Cargo.lock")
+                                                .status()
+                                                .unwrap_or_else(|e| {
+                                                    panic!("could not run git diff-index: {}", e);
+                                                });
+                   if !ret_code.success() {
+                       println!("dirty lock file found at {:?}/Cargo.lock, please run:", path);
+                       println!("\n\tcargo update --manifest-path {:?}/Cargo.lock -p rustc", path);
```

Comment:
Also the `-p` here I think may not be applicable for the standard library, but dropping the `-p` should be fine for now I think